specifyweb/specify/tree_views.py

```diff
@@ -342,4 +343,361 @@ def perm_target(tree):
+        tree_extras.set_fullnames(tree_def_model, null_only=False, node_number_range=None)
+
+        logger.info(f"Added new tree rank: {new_rank_name} with ID: {new_rank_id}")
+        return HttpResponse("Success")
+    
+    except Exception as e:
+        logger.error(f"Error in adding tree rank: {str(e)}")
+        return HttpResponseServerError(f"Failed: {str(e)}")
+
+@openapi.schema({
+    "post": {
+        "parameters": [
+            {
+                "name": "rankName",
+                "in": "formData",
+                "required": True,
+                "schema": {
+                    "type": "string"
+                },
+                "description": "The name of the rank to delete"
+            },
+            {
+                "name": "treeType",
+                "in": "formData",
+                "required": True,
+                "schema": {
+                    "type": "string",
+                    "enum": ["taxon", "geography", "storage", "geologicTimePeriod", "lithoStrat"]
+                },
+                "description": "The type of the tree (taxon, geography, or storage)"
+            }
+        ],
+        "responses": {
+            "200": {
+                "description": "Rank successfully deleted",
+                "content": {
+                    "application/json": {}
+                }
+            },
+            "404": {
+                "description": "Rank not found"
+            },
+            "500": {
+                "description": "Internal server error"
+            }
+        }
+    }
+})
+@tree_mutation
+@require_POST
+def delete_tree_rank(request, tree) -> HttpResponse:
+    """
+    Deletes a rank from the specified tree. Expects 'rank_id' in the POST data.
+    """
+    try:
+        # Get parameter values from request
+        tree_type = request.POST.get('treeType')
+        rank_name = request.POST.get('rankName')
+        
+        # Throw exceptions if the required parameters are not given correctly
+        if rank_name is None:
+            raise Exception("Rank name is not given")
+        if tree_type is None or tree_type.lower() not in TREE_RANKS_MAPPING.keys():
+            raise Exception("Invalid tree type")
+
+        # Get tree model
+        tree_model = getattr(spmodels, tree_type.lower().title())
+
+        # Get tree def item model
+        tree_def_model_name = str.join(tree_type, 'treedef').lower().title()
+        tree_def_item_model_name = str.join(tree_type, 'treedefitem').lower().title()
+        tree_def_model = getattr(spmodels, tree_def_model_name)
+        tree_def_item_model = getattr(spmodels, tree_def_item_model_name)
+
+        # Make sure no nodes are present in the rank before deleting rank
+        rank = tree_def_item_model.objects.get(name=rank_name)
+        nodes_in_rank = tree_model.objects.filter(rank_id=rank.rank_id)
+        if nodes_in_rank is None or nodes_in_rank.count() > 0:
+            raise Exception("The Rank is not empty, cannot delete!")
```

Comment:
should this be something like `raise HttpBadRequest(...)` or something like that so that the front-end could catch this specific case and handle it gracefully (by proving a localized friendly error message to the user)